tests/binary_tests/puppet.test.js

```diff
@@ -57,22 +57,29 @@ function makeAOI() {
 // errors are not thrown from an async beforeAll
 // https://github.com/facebook/jest/issues/8688
 beforeAll(async () => {
+  // start the invest app and forward stderr to console
   electronProcess = spawn(
+    // remote-debugging-port is a chromium arg
     `"${binaryPath}"`, [`--remote-debugging-port=${PORT}`],
     { shell: true }
   );
   electronProcess.stderr.on('data', (data) => {
     console.log(`${data}`);
   });
+
+  // get data about the remote debugging endpoint
   // so we don't make the next fetch too early
-  await new Promise((resolve) => setTimeout(resolve, 5000));
+  await new Promise(resolve => setTimeout(resolve, 20000));
   const res = await fetch(`http://localhost:${PORT}/json/version`);
   const data = JSON.parse(await res.text());
+
+  // connect to the debugging endpoint
   browser = await puppeteer.connect({
```

Comment:
not sure I understand this, `browser` is connected to the debugger URL but it's also used in the test like the actual app?

Reply:
I'm not sure I understand it much better. `puppeteer.connect` is an alternative to `puppeteer.launch`, for starting the browser. A more typical chrome use-case would be `launch` chrome, init a new `page` and then goto a URL in that page and do stuff. But since we don't want to launch a plain chromium browser, we need to first launch our electron app, and then use puppeteer to connect to it, which it somehow does using the devtools debugging protocols.